presto-hive/src/main/java/com/facebook/presto/hive/HiveBucketing.java

```diff
@@ -291,13 +295,22 @@ private static int hashBytes(int initialValue, Slice bytes)
             return Optional.empty();
         }
 
-        Optional<Map<ColumnHandle, NullableValue>> bindings = TupleDomain.extractFixedValues(effectivePredicate);
+        /*Optional<Map<ColumnHandle, NullableValue>> bindings = TupleDomain.extractFixedValues(effectivePredicate);
         if (!bindings.isPresent()) {
             return Optional.empty();
         }
         OptionalInt singleBucket = getHiveBucket(table, bindings.get());
         if (singleBucket.isPresent()) {
             return Optional.of(new HiveBucketFilter(ImmutableSet.of(singleBucket.getAsInt())));
+        }*/
+
+        Optional<Map<ColumnHandle, Set<NullableValue>>> columnValueSetBindings = TupleDomain.extractFixedValuesFromRanges(effectivePredicate);
```

Comment:
columnValueSetBindings -> bindings for brevity

Reply:
Done